### PR TITLE
Compare equality of boards

### LINKED ISSUE #10: Rewrite tests by comparing equality of boards
Rewrite the tests using an initial board and a final board with which to compare, for this apply the concept of equality of two objects

### PR DESCRIPTION
**Pull request by apojomovsky**
Fixes #10 
- Redefines __eq__ for the GameBoard class
- Adds tests for the new __eq__ method
- Re-written player's tests with the new concept of equality

### CODE REVIEW

File: tests/test_gameboard.py [hunk 1]
```diff
@@ -120,3 +120,55 @@ def test_winner_exists_from_diagonals_on_right_corner(self):
         self.assertEqual(self.board_test_diagonals.winner_exists(), False)
         self.board_test_diagonals.put_chip(7, 'red')
         self.assertEqual(self.board_test_diagonals.winner_exists(), True)
+
+    def test_equality_of_empty_boards(self):
+        board_1 = GameBoard.from_matrix([
```
Comment (basicNew) [inline tests/test_gameboard.py]: Very minor thing: you could have an `empty()` class method to do this (`GameBoard.empty()`)

File: tests/test_gameboard.py [hunk 2]
```diff
@@ -120,3 +120,55 @@ def test_winner_exists_from_diagonals_on_right_corner(self):
...
+                    ['red',   'red',  'red', 'blue',  'red', 'blue', None]])
+        board_2 = GameBoard.from_matrix([
+                    ['blue', 'blue', 'blue',  'red',   None,   None, None],
+                    ['red',   'red',  'red', 'blue', 'blue',   None, None],
+                    ['blue',  'red', 'blue',  'red', 'blue',   None, None],
+                    ['blue',  'red', 'blue',  'red',  'red',  'red', None],
+                    ['blue', 'blue',  'red', 'blue',  'red', 'blue', None],
+                    ['red',   'red',  'red', 'blue',  'red', 'blue', None]])
+        self.assertTrue(board_1 == board_2)
+
```
Comment (basicNew) [inline tests/test_gameboard.py:145]: Extra CRs

File: tests/test_gameboard.py [hunk 3]
```diff
@@ -120,3 +120,55 @@ def test_winner_exists_from_diagonals_on_right_corner(self):
...
+                    ['blue', 'blue',  'red', 'blue',  'red', 'blue', None],
+                    ['red',   'red',  'red', 'blue',  'red', 'blue', 'red']])
+        board_2 = GameBoard.from_matrix([
+                    ['blue', 'blue', 'blue',  'red',   None,   None,  None],
+                    ['red',   'red',  'red', 'blue', 'blue',   None,  None],
+                    ['blue',  'red', 'blue',  'red', 'blue',   None,  None],
+                    ['blue',  'red', 'blue',  'red',  'red',  'red',  None],
+                    ['blue', 'blue',  'red', 'blue',  'red', 'blue',  None],
+                    ['red',   'red',  'red', 'blue',  'red', 'blue', 'blue']])
+        self.assertFalse(board_1 == board_2)
```
Comment (basicNew) [inline tests/test_gameboard.py]: Another interesting case to test would be that of having the actual same matrix (remember equality vs identity):

```
matrix = [...]
board_1 = GameBoard.from_matrix(matrix)
board_2 = GameBoard.from_matrix(matrix)
```

Comment (basicNew) [reply tests/test_gameboard.py]: If you are testing equality you should also test for `!=`. See http://stackoverflow.com/questions/390250/elegant-ways-to-support-equivalence-equality-in-python-classes for a more detailed discussion on the subject (we will deal with hashes later on, so for now just take a look up to "Are we there yet? Not quite. How many unique numbers do we have?")

**Review by basicNew — COMMENTED** [review]
Looking good @apojomovsky , just left a couple of comments

**Comment by apojomovsky** [comment]
addressed all your comments, is ready for another pass

**Pull request merged**